bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/module/handler/AnnotationActionHandler.java

```diff
@@ -136,16 +144,17 @@ public AnnotationActionHandler(Action module, ActionType mt, Method method, Obje
                             method, moduleType.getUID(), ex.getMessage());
                 }
                 // we allow simple data types as return values and put them under the context key "result".
-            } else if (result instanceof Boolean booleanValue) {
-                output.put(MODULE_RESULT, booleanValue);
-            } else if (result instanceof String) {
+            } else if (result instanceof Boolean || result instanceof String || result instanceof Byte
```

Comment:
Don‘t we need the casting here so we don’t return a Object?

Reply:
Vales looks good in the UI pop-up.
But I can cast.

Reply:
Cast done

Reply:
When sent via REST, the value are serialised, for that it doesn’t matter.
But when used as rule action module, I think it matters as the value is passed along and not serialised, so thanks for adding casting.